python/revng/cli/_commands/idb_converter.py

```diff
@@ -0,0 +1,655 @@
+
+            # Empty struct are illegal -- emit them as a typedef void <struct>.
+            elif len(type.type_details.members) == 0:
+                logger.warning(
+                    f"Found invalid empty struct {type_name}, replacing with a void* typedef"
+                )
+                underlying_type = self._get_primitive_type(m.PrimitiveTypeKind.Void, 0)
+                underlying_type.Qualifiers.append(
+                    m.Qualifier(
+                        Kind=m.QualifierKind.Pointer,
+                        Size=8 if self.is64bit else 4,
+                    )
+                )
+                revng_type = m.TypedefType(OriginalName=type_name, UnderlyingType=underlying_type)
+            else:
+                # Struct members and size will be computed later.
+                revng_type = m.StructType(OriginalName=type_name, Size=0, Fields=[])
+                self._structs_to_fixup.add((revng_type, type))
+
+        elif type.is_decl_union():
+            # Union members will be computed later.
+            revng_type = m.UnionType(OriginalName=type_name, Fields=[])
+            self._unions_to_fixup.add((revng_type, type))
+
+        elif type.is_decl_ptr():
+            underlying_type = self._convert_idb_type_to_revng_type(type.type_details.obj_type)
+            revng_type = self.resolve_typeref(underlying_type.UnqualifiedType)
+            revng_type_qualifiers = list(underlying_type.Qualifiers)
+
+        elif type.is_decl_array():
+            underlying_type = self._convert_idb_type_to_revng_type(type.type_details.elem_type)
+            revng_type = self.resolve_typeref(underlying_type.UnqualifiedType)
+            revng_type_qualifiers = list(underlying_type.Qualifiers)
+
+        elif type.is_decl_bool():
+            size = type.get_size()
+            revng_type = self.unwrap_qualified(
+                self._get_primitive_type(m.PrimitiveTypeKind.Unsigned, size)
+            )
+
+        elif type.is_decl_int() or type.is_decl_floating():
+            size = type.get_size()
+            primitive_kind = get_primitive_kind(type)
+            revng_type = self.unwrap_qualified(self._get_primitive_type(primitive_kind, size))
+
+        elif type.is_decl_void():
+            primitive_kind = m.PrimitiveTypeKind.Void
+            size = 0
+            if type.get_name() != "":
+                # Treat this case as `typedef void someothername`.
+                revng_void_type = self._get_primitive_type(primitive_kind, size)
+                revng_type = m.TypedefType(OriginalName=type_name, UnderlyingType=revng_void_type)
+            else:
+                revng_type = self.unwrap_qualified(self._get_primitive_type(primitive_kind, size))
+
+        elif type.is_decl_func():
+            # TODO: handle non C-ABI functions.
+            # We cannot handle stack arguments at the moment.
+            assert type.type_details.stkargs is None
```

Comment:
I was playing a bit with this, and my conclusion is that the ```python-idb``` does not populate/use this field at all. Also, I am seeing that (as far as I can tell), the IDA is not very good when recovering signatures of the functions with the stack arguments, so that may be the reason the field was not used at all in the package.

Reply:
Are you 100% sure? x86 32-bit only uses stack arguments, and IDA has been supporting that forever.

Reply:
Hmm, good example. I was playing with some very complex examples, but I made some very simple ones and IDA is able to recognize the signature. But anyhow, the ```type.type_details.stkargs``` are still ```None``` and all the arguments are being populated within ```type.type_details.args``` (for both stack and register arguments), so I guess we should just refactor the ```TODO``` by saying something like "This field is unused in the package....".

Reply:
OK. So we *do* support importing stack arguments, correct?
Up to you whether to drop the code entirely or leave a TODO.

Reply:
Right, we *do* support them.